plugins/camera/camera_impl.cpp

```diff
@@ -782,10 +782,15 @@ CameraImpl::to_euler_angle_from_quaternion(Camera::CaptureInfo::Quaternion quate
 
 void CameraImpl::notify_capture_info(Camera::CaptureInfo capture_info)
 {
-    if (_capture_info.callback) {
-        _parent->call_user_callback(
-            [this, capture_info]() { _capture_info.callback(capture_info); });
+    // Make a copy because it is passed to the thread pool
+    const auto capture_info_callback = _capture_info.callback;
+
+    if (capture_info_callback == nullptr) {
+        return;
     }
+
+    _parent->call_user_callback(
+        [capture_info, capture_info_callback]() { capture_info_callback(capture_info); });
```

Comment:
I'm not completely sure how the copy behaves for the capture by-value of `capture_info`... :thinking:. Do you have an opinion @julianoes?